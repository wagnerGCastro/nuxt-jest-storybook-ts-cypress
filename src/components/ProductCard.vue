<template>
  <v-card class="card-product">
    <v-img lazy-src="https://picsum.photos/id/11/10/6" :src="product.image"></v-img>

    <v-card-actions class="vcard-actions">
      <v-btn class="cart-btn" color="primary" nuxt @click.prevent="addToCart">
        <v-icon dense class="icon">mdi-cart-outline</v-icon>
      </v-btn>
    </v-card-actions>

    <v-card-title class="headline">{{ product.title }}</v-card-title>
    <v-card-text>R$ {{ product.price }}</v-card-text>
  </v-card>
</template>

<script>
import { mapActions } from 'vuex';

export default {
  name: 'ProductCard',

  props: {
    product: {
      type: Object,
      default: () => ({
        id: {
          type: [String, Number],
          default: '',
        },
        title: {
          type: String,
          default: 'Teste Product',
        },
        price: {
          type: [String, Number],
          default: '0.00',
        },
        image: {
          type: String,
          default: 'https://picsum.photos/id/11/500/300',
        },
      }),
    },
  },
  data: () => ({}),
  computed: {},
  async created() {},
  methods: {
    ...mapActions({ addProduct: 'cart/add' }),

    addToCart() {
      this.addProduct(this.product);
    },
  },
};
</script>

<style scoped>
.card-product .v-card__title.headline {
  font-size: 15px !important;
  line-height: 1.3rem;
  text-transform: uppercase;
  padding-top: 1px !important;
  padding-bottom: 5px !important;
}

.card-product .vcard-actions {
  margin: -30px -41px 0 78%;
  display: flex;
  justify-content: center;
  max-width: 40px;
}

.card-product .v-btn.cart-btn {
  height: 36px;
  width: 36px;
  border-radius: 50%;
}

.card-product .v-btn.cart-btn svg {
  height: 20px;
}

.card-product .v-btn:not(.v-btn--round).v-size--default {
  min-width: auto;
}
</style>
