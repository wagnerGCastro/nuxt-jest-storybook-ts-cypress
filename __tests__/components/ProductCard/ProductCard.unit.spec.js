import Vuetify from 'vuetify';
import { mount } from '@vue/test-utils';
import ProductCard from '@/components/ProductCard';

// const vuetify = new Vuetify();

describe('ProductCard - unit', () => {
  let vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  fit('should mount the component', () => {
    const wrapper = mount(ProductCard, {
      vuetify,
      propsData: { product: {} },
    });

    console.log(wrapper.html());

    expect(wrapper.vm).toBeDefined();
  });
});
