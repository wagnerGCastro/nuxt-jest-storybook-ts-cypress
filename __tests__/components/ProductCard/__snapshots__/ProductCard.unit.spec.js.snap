// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard - unit should match snapshot 1`] = `
<div
  class="card-product v-card v-sheet theme--light"
>
  <div
    class="v-image v-responsive theme--light"
  >
    <transition-stub
      mode="in-out"
      name="fade-transition"
    >
      <div
        class="v-image__image v-image__image--preload v-image__image--cover"
        style="background-image: url(https://picsum.photos/id/11/10/6); background-position: center center;"
      />
    </transition-stub>
    <div
      class="v-responsive__content"
    />
  </div>
   
  <div
    class="v-card__actions vcard-actions"
  >
    <button
      class="cart-btn v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default primary"
      type="button"
    >
      <span
        class="v-btn__content"
      >
        <i
          aria-hidden="true"
          class="v-icon notranslate icon v-icon--dense mdi mdi-cart-outline theme--light"
        />
      </span>
    </button>
  </div>
   
  <div
    class="v-card__title headline"
  >
    Rel√≥gio Bonito
  </div>
   
  <div
    class="v-card__text"
  >
    R$ 23.00
  </div>
</div>
`;
